src/library/scala/collection/mutable/ArraySeq.scala

```diff
@@ -184,6 +184,27 @@ object ArraySeq extends StrictOptimizedClassTagSeqFactory[ArraySeq] { self =>
       case that: ofChar => Arrays.equals(array, that.array)
       case _ => super.equals(that)
     }
+
+    override def addString(sb: StringBuilder, start: String, sep: String, end: String): StringBuilder = {
+      val jsb = sb.underlying
+      val len = array.length
+      jsb.append(start)
+      if (len > 0) {
+        if (sep.isEmpty) jsb.append(array)
```

Comment:
This doesn't look right!  `array.toString` doesn't produce the concatenated `toString` of its components.

Reply:
This is using append(<Array[Char]>)

Reply:
Oh, I missed the context.  Whoops!